src/Core/Config/ConfigLoader.php

```diff
@@ -78,14 +82,31 @@ public function countManifests()
     }
 
     /**
-     * Nest the current manifest
+     * Nest the config loader and activates it
      *
-     * @return ConfigCollectionInterface
+     * @return static
      */
     public function nest()
     {
-        $manifest = $this->getManifest()->nest();
-        $this->pushManifest($manifest);
-        return $manifest;
+        // Nest config
+        $manifest = clone $this->getManifest();
+
+        // Create new blank loader with new stack (top level nesting)
+        $newLoader = new self;
```

Comment:
In `activate()` we're using `static` - this won't consider late static binding - should it?